system/metrics/sinks/elastic/elasticSink.cpp

```diff
@@ -43,8 +42,66 @@ ElasticMetricSink::ElasticMetricSink(const char *name, const IPropertyTree *pSet
     PeriodicMetricSink(name, "elastic", pSettingsTree)
 {
     ignoreZeroMetrics = pSettingsTree->getPropBool("@ignoreZeroMetrics", true);
-    pSettingsTree->getProp("@elasticHost", elasticHost);
-    pSettingsTree->getProp("@indexName", indexName);
+
+    StringBuffer hostName;
+    StringBuffer hostProtocol;
+    StringBuffer hostPort;
+
+    Owned<IPropertyTree> pHostConfigTree = pSettingsTree->getPropTree("host");
+    if (pHostConfigTree)
+    {
+        pHostConfigTree->getProp("@name", hostName);
+
+        if (!pHostConfigTree->getProp("@protocol", hostProtocol))
+        {
+            hostProtocol.append("https");
+        }
+
+        if (!pHostConfigTree->getProp("@port", hostPort))
+        {
+            hostPort.append("9200");
+        }
+    }
+
+    if (!hostName.isEmpty() && !hostPort.isEmpty() && !hostProtocol.isEmpty())
+    {
+        elasticHostUrl.append(hostProtocol).append("://").append(hostName).append(":").append(hostPort);
+    }
+    else
+    {
+        WARNLOG("ElasticMetricSink: Host configuration missing or invalid");
+    }
+
+    Owned<IPropertyTree> pIndexConfigTree = pSettingsTree->getPropTree("index");
+    if (pIndexConfigTree)
+    {
+        pSettingsTree->getProp("@name", indexName);
+    }
+
+    if (indexName.isEmpty())
+    {
```

Comment:
style: We do not normally use {} if an if statement only has a single statement - here and in several other places in this file.  It makes it slightly harder to read.